### PR TITLE
Implement prismarine-registry, basic block entities and 1.18 disk loading

### PR DESCRIPTION
**Pull request by extremeheat**
depends on prismarine-block update

WIP, this goes along with updating anvil-provider for 1.18 level format

### CODE REVIEW

File: src/bedrock/0.14/chunk.js [hunk 1]
```diff
@@ -17,13 +17,13 @@ const writeUInt4LE = require('uint4').writeUInt4LE
 
 module.exports = loader
 
-function loader (mcVersion) {
-  Block = require('prismarine-block')(mcVersion)
+function loader (registry) {
```
Comment (rom1504) [inline src/bedrock/0.14/chunk.js:20]: same point as in prismarine-block, let's support both registry and version here

File: src/pc/1.15/ChunkColumn.js [hunk 2]
```diff
@@ -190,10 +195,6 @@ module.exports = (Block, mcData) => {
       this.biomes[getBiomeIndex(pos)] = biome
     }
 
-    setBiomeColor (pos, r, g, b) {
```
Comment (rom1504) [inline src/pc/1.15/ChunkColumn.js:193]: this needs to stay in existence due to the fact it exists in some mcpe (unless you remove these versions completely)

File: src/pc/common/PaletteContainer.js [hunk 3]
```diff
@@ -127,7 +127,7 @@ class IndirectPaletteContainer {
 class SingleValueContainer {
   constructor (options) {
     this.value = options?.value ?? 0
-    this.bitsPerValue = options?.bitsPerValue ?? constants.MIN_BITS_PER_BLOCK
```
Comment (nickelpro) [inline src/pc/common/PaletteContainer.js:130]: This is wrong, this value represents the `bitsPerValue` to use when _upgraded_ to a non-single value container. The default can be whatever, but you _must_ respect the option when passed.

Comment (extremeheat) [reply src/pc/common/PaletteContainer.js]: Fixed

File: src/pc/1.17/ChunkColumn.js [hunk 4]
```diff
@@ -153,11 +163,6 @@ module.exports = (Block, mcData) => {
       return this.biomes[getBiomeIndex(pos, this.minY)]
     }
 
-    getBiomeColor (pos) {
```
Comment (rom1504) [inline src/pc/1.17/ChunkColumn.js:156]: can you undo this?

File: src/pc/1.15/ChunkColumn.js (hunk 2 above)
Comment (nickelpro) [reply src/pc/1.15/ChunkColumn.js:193]: They still exist, they've just been moved to `CommonChunkColumn` which this extends

File: src/pc/1.17/ChunkColumn.js (hunk 4 above)
Comment (nickelpro) [reply src/pc/1.17/ChunkColumn.js:156]: Did here, moved to `CommonChunkColumn`

File: src/pc/1.15/ChunkColumn.js (hunk 2 above)
Comment (extremeheat) [reply src/pc/1.15/ChunkColumn.js:193]: only the pc/* versions have the getBiomeColor/setBiomeColor moved to pc/common/CommonChunkColumn, so mcpe versions shouldn't be effected

**Comment by rom1504** [comment]
bumped pblock

**Comment by extremeheat** [comment]
Entity data should go into `.entity` property of a prismarine-block passed to getBlock/setBlock. Added some doc here for the `getBlockEntity(pos)` and `setBlockEntity(pos, nbt)` functions. If the block entities are sent along with the chunk data it should be possible to add in a extra parameter to the load() function, otherwise it can use loadBlockEntities(nbtArray)... 

Now since it looks like the "id" string field isn't written on the chunk packet that may have to be added in somewhere

**Comment by rom1504** [comment]
ok sounds good for loadBlockEntities
what about dump ? I guess there should be a dumpBlockEntities ?

**Comment by rom1504** [comment]
ok sounds good for loadBlockEntities
what about dump ? I guess there should be a dumpBlockEntities ?

**Comment by rom1504** [comment]
ok sounds good for loadBlockEntities
what about dump ? I guess there should be a dumpBlockEntities ?

**Comment by rom1504** [comment]
I'm thinking it would make sense to try to integrate this in mineflayer before merging
so we don't need to change the API after merging

**Comment by nickelpro** [comment]
We could major version bump here, and then integrate to mineflayer and make any tweaks that come up. That way mineflayer maintains continuity of function.

**Comment by extremeheat** [comment]
For mineflayer I think moving most of the relevant world data to prismarine-world would make sense to do now, to avoid things having to be reworked a second time

**Comment by rom1504** [comment]
> We could major version bump here, and then integrate to mineflayer and
make any tweaks that come up. That way mineflayer maintains continuity of
function

My point was to avoid doing many major versions of prismarine-chunk because
we don't know what we need before integrating

**Comment by rom1504** [comment]
> For mineflayer I think moving most of the relevant world data to
prismarine-world would make sense to do now, to avoid things having to be
reworked a second time

Yes let's do it.
I believe it will be cleaner to do it before we merge this. (Just do one PR
for mineflayer one for pworld and one for pchunk (this one))

**Comment by nickelpro** [comment]
Sounds like a plan, would it be wise to move search at the same time? I can open a PR off this branch for it

**Comment by rom1504** [comment]
missing chunk.dumpBlockEntities

**Pull request merged**